src/serac/physics/solid_functional.hpp

```diff
@@ -608,6 +788,96 @@ class SolidFunctional<order, dim, Parameters<parameter_space...>, std::integer_s
   const serac::FiniteElementDual& nodalForces() { return nodal_forces_; };
 
 protected:
+  /**
+   * @brief Rotate the mesh nodes, displacement, and velocity grid functions to align with the sliding wall boundary
+   * condition
+   */
+  void rotate()
+  {
+    // Check if a sliding wall boundary condition exists
+    if (coordinate_transform_) {
+      // Rotate the mesh nodes
+      auto* nodes = mesh_.GetNodes();
+      rotateGridFunction(*coordinate_transform_, *nodes);
+
+      // Get the displacement grid function from the state
+      auto& displacement_gf = displacement_.gridFunction();
+      // Rotate the displacement
+      rotateGridFunction(*coordinate_transform_, displacement_gf);
+      // Set the underlying true vector from the rotated grid function
+      displacement_.setFromGridFunction(displacement_gf);
+
+      // Rotate the velocity
+      auto& velocity_gf = velocity_.gridFunction();
+      rotateGridFunction(*coordinate_transform_, velocity_gf);
+      velocity_.setFromGridFunction(velocity_gf);
+
+      // Rotate the adjoint displacement
+      auto& adjoint_disp = adjoint_displacement_.gridFunction();
+      rotateGridFunction(*coordinate_transform_, adjoint_disp);
+      adjoint_displacement_.setFromGridFunction(adjoint_disp);
+    }
```

Comment:
these rotate and undoRotate functions worry me for a few reasons:
- they seem pretty expensive (manually touching every single dof of many different fields)
- they're done frequently 
- the underlying implementation (rotateGridFunction) is CPU-specific

Reply:
Do you think this would be better in the residual calculation? The only problem is then modifying the stiffness would get complicated.